testflight/fixtures/task_vars.yml

```diff
@@ -42,3 +49,16 @@ jobs:
     vars:
       image_resource_type: mock
       echo_text: ((.:foo))
+
+- name: task-var-is-not-used
```

Comment:
Minor nit: this test (and the underlying issue) isn't really about task vars not being used - it's about referencing both task variables and `var_source` (or global credential manager) variables at the same time in a task. Could you rename this job and the associated test accordingly?

Reply:
The point of this task is, `task: run` takes a var named `foo` that is not used in the task file, which simulates #5753. So I think the job name reflects to goal of this task.

Reply:
Hmm, what I'm struggling with though is that if we were to either get rid of the unused variable `foo` or reference `foo` in the task, this test case would still capture the issue in #5753 (i.e. fail on master and pass with this fix). So, this isn't really about testing an unused variable

Reply:
my point being, unused variables in task vars aren't the issue. For instance, consider the following task:

```yaml
platform: linux
image_resource:
  type: registry-image
  source: {repository: busybox}
run:
  path: echo
  args:
  - ((A))
```

with the following pipeline:

```yaml
...
jobs:
- name: job
  plan:
  - get: repo
  - task: something
    file: repo/tasks/echo-with-vars.yml
    vars:
      A: this is used
      B: this is not used
```

This will work on without this bug fix.

Reply:
Okey, I got your point. Just renamed the task.